### ISSUE TITLE
More flexible attribute customization

### ISSUE THREAD

**Issue by cont-amibabkin** [Esri/calcite-design-system#12456]
### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

We use a custom menu consisting of a `calcite-action` (opens the menu), a `calcite-popover` and `calcite-button's` (menu items).

1. We would like to be able to change the role of the button that is in the shadow root. We need to set role="menuitem" for the button. We also use a button that should have a role="combobox ".

2. We would like to be able to change the role of the `calcite-popover` or make it empty. We need to set the menu role to either the popover or the our container inside the popover.

3. We would like to be able to set the `aria-haspopup`, `aria-controls`, `aria-expanded` attributes ourselves on the button that is in the shadow root (this button is inside the `calcite-action`). These attributes allow the screen reader to tell the user that the button opens a menu.

4. The problem of the type of `calcite-action` has already been created.

### Actual Behavior

![Image](https://github.com/user-attachments/assets/f8c2a13a-f425-4c30-b124-f9c2f708ac8d)

### Expected Behavior

![Image](https://github.com/user-attachments/assets/599deb63-b1c4-4377-9d38-46ac3026f748)

### Reproduction Sample

https://codepen.io/Andrey-B-the-animator/pen/ZYGdJpP

### Reproduction Steps

1. Enable a screen reader.
2. Navigate to the action and activate it.
3. Navigate to the buttons in the popover and notice how the elements are declared.

### Reproduction Version

3.0.3

### Working W3C Example/Tutorial

https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Business/Community Analyst

**Comment by macandcheese** [comment]
Specifically regarding use of buttons there…

It could be worth exploring using Menu and Menu Item to achieve this, which already have these roles. Additionally, a set of three buttons like that will not be keyboard navigable with arrows as a user may expect them to be - Menu Items / Menu have this built in.

**Comment by cont-amibabkin** [comment]
@macandcheese We know this and we already had experience in independently setting up navigation using arrows in custom elements. I believe that we can do navigation ourselves according to the documentation. At the moment we only need to be able to set attributes.

**Comment by isaacbraun** [comment]
Hi Andrey,

To emphasize Adam's point, here is [an example](https://codepen.io/isaacbraun/pen/gbaYYrv?editors=1100) of using the Menu/Menu Item components with some example styling that may fulfill your requirements.

At this time, we are not able to support role changes for any of our components. If possible, we recommend using the Menu/Menu Item components with custom styling. If no other components can be utilized, it may be best to create a custom implementation and utilize Calcite CSS Tokens for styling to ensure a consistent design.

For these reasons, I am closing this issue, but please reach out if you have any questions.
